js/components/common/react_modal.js.jsx

```diff
@@ -34,19 +34,86 @@ class CourseModal extends React.Component {
       course: [],
       sessions: [],
       courseTitle: "",
+      clicked: 0,
     }
   }
+  /**
+   * Change the clicked and courseId state, whenever a course link is clicked.
+   */
+  clickRelated = s => {
```

Comment:
Pick a better parameter name. In general do not use single-letter variable names.

Also, the function name should describe what the function does, not where it's going to be used. "click" is something the user does, but this function name should describe *what happens when a course name is clicked*.